src/clarity-angular/layout/tabs/tabs.ts

```diff
@@ -3,35 +3,84 @@
+import {TabLinkDirective} from "./tab-link.directive";
 
-// TODO: find a better way to separately render the tab contents (i.e. not inside the ul)
 @Component({
     selector: "clr-tabs",
-    template: `
+    template: `        
         <ul class="nav" role="tablist">
-            <ng-content select="clr-tab"></ng-content>
+            <ng-container #inlineTabLinks></ng-container>
+            <!--this is the part that goes into the overflow-->
+            <ng-container *ngIf="tabsService.overflowTabs.length > 0">
+                <div class="tabs-overflow bottom-right" [class.open]="ifOpenService.open" 
+                     (click)="toggleOverflow($event)">
+                    <li role="presentation" class="nav-item">
+                        <button class="btn btn-link nav-link dropdown-toggle" [class.active]="activeTabInOverflow">
+                            <clr-icon shape="ellipsis-horizontal" [class.is-info]="ifOpenService.open"></clr-icon>
+                        </button>
+                    </li>
+                    <clr-tab-overflow-content>
+                        <ng-container #overflowTabLinks></ng-container>
+                    </clr-tab-overflow-content>
+                </div>
+            </ng-container>
         </ul>
+        <!--this is the tab content, rendered outside the ul element above-->
+        <!--TODO: find a better way to achieve this-->
         <ng-container *ngFor="let tab of tabsService.children">
             <ng-template *ngIf="tab.tabContent" [ngTemplateOutlet]="tab.tabContent.templateRef"></ng-template>
         </ng-container>
     `,
-    providers: [ IfActiveService, TabsService ]
+    providers: [ IfActiveService, IfOpenService, TabsService ]
 })
 export class Tabs {
-    constructor(public ifActiveService: IfActiveService, public tabsService: TabsService) {
+    private _prevOverflowTabs: Tab[] = [];
+    @ViewChild("inlineTabLinks", {read: ViewContainerRef}) inlineTabLinksContainerRef: ViewContainerRef;
+    @ViewChild("overflowTabLinks", {read: ViewContainerRef}) overflowTabLinksContainerRef: ViewContainerRef;
+    @ContentChildren(TabLinkDirective, {descendants: true}) tabLinkDirectives: QueryList<TabLinkDirective>;
+
+    constructor(public ifActiveService: IfActiveService,
+                public ifOpenService: IfOpenService,
+                public tabsService: TabsService) {
+    }
+
+    get activeTabInOverflow() {
+        return this.tabsService.overflowTabs.indexOf(this.tabsService.activeTab) > -1;
     }
 
-    get activeTab() {
-        let activeTab: Tab;
-        if (this.tabsService.children.length > 0) {
-            activeTab = this.tabsService.children.find((tab: Tab) => {
-                return tab.active;
-            });
+    ngAfterViewInit() {
+        this.renderTabLinks();
+    }
+
+    ngAfterViewChecked() {
+        if (this.overflowTabsChanged()) {
+            this.renderTabLinks();
+            this._prevOverflowTabs = this.tabsService.overflowTabs;
```

Comment:
I believe you now have a problem with dynamic inline tabs. They do not update properly.

Rather than address this manually, I would like you to use a behavior closer to the button groups one: only trigger a redraw when a tab link's `inOverflow` input actually changes.
The reason behind this change request is that I'm a bit scared of the potential performance impact in the future if we let a longer operation slip in the `if` check.